### DISCUSSION TITLE
skOpen 与pktRx,pktTx的关系是什么

### DISCUSSION THREAD

**Discussion by fishyu-mushroom** [baidu/dperf#424] [Q&A]
你好，官方提供的报告中 skOpen  与pktRx,pktTx基本上是1:3的关系，
seconds 22                 cpuUsage 52
pktRx   3,001,058          pktTx    3,001,025          bitsRx   2,272,799,040      bitsTx  1,920,657,600      dropTx  0
arpRx   0                  arpTx    0                  icmpRx   0                  icmpTx  0                  otherRx 0          badRx 0
synRx   1,000,345          synTx    1,000,330          finRx    1,000,350          finTx   1,000,350          rstRx   0          rstTx 0
synRt   0                  finRt    0                  ackRt    0                  pushRt  0                  tcpDrop 0
skOpen  1,000,330          skClose  1,000,363          skCon    230                skErr   0
httpGet 1,000,345          http2XX  1,000,350          httpErr  0
ierrors 0                  oerrors  0                  imissed  0
我使用dperfV1.6版本，不管怎么调配置文件cps都是     skOpen  与pktRx,pktTx 关系是1:1的样子？请问这个原因是什么？

Dperf Client 
mode            client
cpu               0 1 2 3 4 5 6 7
slow_start     60
tx_burst        128
launch_num 10
payload_size 1
duration        90s
protocol        tcp
cps                [refer to performance data]
port               0000:82:00.0    10.26.3.120    10.26.3.254
client             10.26.3.120    50
server            10.26.3.12      8
listen             80                 1

Dperf Server
mode               server
cpu                  0 1 2 3 4 5 6 7
tx_burst           128
payload_size   1
duration          100d
port                 0000:82:00.0 10.26.3.21 10.26.3.254
client               10.26.3.120   100
server              10.26.3.21     8
listen               80                 1

**Comment by pengjianzhang** [comment]
与测试配置参数相关，长连接、短链接、keepalive等，不是固定的
如果喜欢dperf 请加个star 谢谢

**Reply by fishyu-mushroom** [reply]
这个和dperf版本没有关系吗？我看官方的文档https://github.com/baidu/dperf/tree/v1.6.0与https://github.com/baidu/dperf/tree/v1.2.0的实验数据都完全一致，这个后续没有再验证吗？
Statistics
dperf outputs various statistics every second：

TPS, CPS, various PPS
Errors of TCP/Socket/HTTP
Packets loss/drop
Retransmissions of TCP Flags
seconds 22                 cpuUsage 52
pktRx   3,001,058          pktTx    3,001,025          bitsRx   2,272,799,040      bitsTx  1,920,657,600      dropTx  0
arpRx   0                  arpTx    0                  icmpRx   0                  icmpTx  0                  otherRx 0          badRx 0
synRx   1,000,345          synTx    1,000,330          finRx    1,000,350          finTx   1,000,350          rstRx   0          rstTx 0
synRt   0                  finRt    0                  ackRt    0                  pushRt  0                  tcpDrop 0
skOpen  1,000,330          skClose  1,000,363          skCon    230                skErr   0
httpGet 1,000,345          http2XX  1,000,350          httpErr  0
ierrors 0                  oerrors  0                  imissed  0

**Comment by pengjianzhang** [comment]
用最新的代码测吧
